openpype/hosts/maya/plugins/publish/validate_frame_range.py

```diff
@@ -53,12 +53,32 @@ def process(self, instance):
             )
             return
 
-        frame_start_handle = int(context.data.get("frameStartHandle"))
-        frame_end_handle = int(context.data.get("frameEndHandle"))
-        handle_start = int(context.data.get("handleStart"))
-        handle_end = int(context.data.get("handleEnd"))
-        frame_start = int(context.data.get("frameStart"))
-        frame_end = int(context.data.get("frameEnd"))
+        # Get to the task doc of the project, folder path and task
+        project_name = instance.context.data["projectEntity"]["name"]
+        folder_path = instance.context.data["asset"]
+        task_name = instance.context.data["task"]
+        from ayon_api import get_task_by_name, get_folder_by_path
+        folder = get_folder_by_path(project_name, folder_path)
+        asset_id = folder["id"]
+        task = get_task_by_name(project_name, asset_id, task_name)
```

Comment:
@BigRoy so would you recommended adding a "CollectTaskEntity" to the Maya host collectors.
I want to store the task doc attributes for later query in validators.

Reply:
Yup, task entity collecting would make sense. It's happening already for `projectEntity` and `assetEntity` or alike on both `context` and `instance` (note that Instance MAY target a different asset than current context and thus might need a validation against that instead of the context).

I'd look up which collector is currently storing `projectEntity` and `assetEntity` which I think it does in a bulk query from a `ContextPlugin` and add the data collection for `taskEntity` there as well (_if_ `task` is set on the instance/context, etc.)

@iLLiCiTiT thoughts?

Reply:
I found where project and asset entity are added to context, this is in 
collect_context_entities.py
I now prepare and cache the "taskEntity"here.
https://github.com/ynput/OpenPype/pull/6207/files#diff-65495e9f1aba1baa5c1f7517ad5f56e80c42990db087260f8ee4a5eee6edb1c1R17
```
task_entity = ayon_api.get_task_by_name(
    project_name, asset_entity["_id"], task_name)
context.data["taskEntity"] = task_entity
```

Reply:
@BigRoy it seems get_task_by_name is only in ayon_api. 

I guess I need to make OpenPype legacy work without ayon in the mix?

In which case I should test if AYON_SERVER_ENABLED, and then only do the query in ayon.

I assume there was no API for this in legacy OpenPype

Reply:
it would be easier if I don't have to back support OpenPype without AYON.

Reply:
Yup, there's no equivalent "task" data in OpenPype as far as I know.

Since this is in the OpenPype repo I'm pretty sure you'll still need to maintain backwards compatibility. But might be a better question forwarded to @mkolar or @iLLiCiTiT